### PR TITLE
Volume collection related changes

### PR DESCRIPTION
**Pull request by AnuBose**
Added volume collection to list under available resources
Added volume collection id to create volume and get volume

### CODE REVIEW

File: examples/data-sources/hpegl_metal_available_resources/README.md [hunk 1]
```diff
@@ -113,6 +113,8 @@ volumes = [
     "size" = 0
     "storage_pool" = "Storage_Pool_NVMe"
     "storage_pool_id" = "8cd530bb-9ac2-4e28-9a6d-b7a695940e46"
+    "volume_collection" = "collection_for_autsin"
```
Comment (patricia-dibenedetto) [inline examples/data-sources/hpegl_metal_available_resources/README.md:116]: minor: typo "austin"

File: examples/resources/hpegl_metal_volume/resource.tf [hunk 2]
```diff
@@ -7,16 +7,16 @@ provider "hpegl" {
 }
 
 variable "location" {
-  default = "USA:Central:AFCDCC1"
+  default = "USA:CO:FTC"
```
Comment (patricia-dibenedetto) [inline examples/resources/hpegl_metal_volume/resource.tf:10]: as I recall, this file is setup to work well with simulated resources, why this change?

File: examples/resources/hpegl_metal_volume/resource.tf [hunk 3]
```diff
@@ -7,16 +7,16 @@ provider "hpegl" {
...
 }
 
 resource "hpegl_metal_volume" "test_vols" {
-  count        = 1
-  name         = "vol-${count.index}"
-  size         = 20
-  shareable    = true
-  flavor       = "Fast"
-  storage_pool = "Storage_Pool_NVMe"
```
Comment (patricia-dibenedetto) [inline examples/resources/hpegl_metal_volume/resource.tf:19]: think I added storage_pool to exercise new feature, why remove it?

File: internal/resources/resource_volume.go [hunk 4]
```diff
@@ -136,6 +138,20 @@ func volumeSchema() map[string]*schema.Schema {
...
 		},
+
+		vCollection: {
+			Type:        schema.TypeString,
+			Required:    false,
+			Optional:    true,
+			Description: "The volume collection of the volume to be created.",
+		},
+
+		vCollectionID: {
```
Comment (patricia-dibenedetto) [inline internal/resources/resource_volume.go:149]: should this be similar to vStoragePoolID? see Computed: true

Comment (AnuBose) [reply internal/resources/resource_volume.go:149]: I am allowing to specify either name or ID. Is that a problem?

Comment (patricia-dibenedetto) [reply internal/resources/resource_volume.go:149]: just saying it differs from other optional ID fields so thought we should be consistent.
Marking as Computed, it can still be provided . Per docs: 
"Computed indicates whether the provider may return its own value for this attribute or not. Computed cannot be used with Required. If Required and Optional are both false, the attribute will be considered "read only" for the practitioner, with only the provider able to set its value."

Comment (AnuBose) [reply internal/resources/resource_volume.go:149]: I am saying, it is not computed because I allow the user to enter the value in.
It provides the function to create volume- either the user enters name or id.
I do not see the need to change this. 
I dont see there is any consistency issue

Comment (patricia-dibenedetto) [reply internal/resources/resource_volume.go:149]: marking it Computed still allows user to enter it if desired, expectation is probably they will use the name (not the id) in which case the id will be computed

Comment (AnuBose) [reply internal/resources/resource_volume.go:149]: it does not make any differenernce sonce the filed is optional anyway. 
If you are bent upon debating this and debating I will change it - I do not see any technical need to change it for the  record .

**Review by patricia-dibenedetto — APPROVED** [review]
